### ISSUE TITLE
Cannot build with shared webcore

### ISSUE THREAD

**Issue by mkaracsony81** [annulen/webkit#586]
Tried to build it with shared_webcore enabled, build fails in final stage (processentries, undefined references).

**Comment by annulen** [comment]
Out of curiosity, why do you need this build configuration?

**Comment by annulen** [comment]
```diff
diff --git a/Source/WebKit2/PlatformQt.cmake b/Source/WebKit2/PlatformQt.cmake
index 5243ef2d812..3a69604b8e7 100644
--- a/Source/WebKit2/PlatformQt.cmake
+++ b/Source/WebKit2/PlatformQt.cmake
@@ -280,25 +280,27 @@ list(APPEND WebProcess_SOURCES
     qt/MainQt.cpp
 )

+if (NOT SHARED_CORE)
+    set(WebProcess_LIBRARIES
+        WebKit
+    )
+    set(NetworkProcess_LIBRARIES
+        WebKit
+    )
+    set(DatabaseProcess_LIBRARIES
+        WebKit
+    )
+    set(PluginProcess_LIBRARIES
+        WebKit
+    )
+endif ()
+
 # FIXME: Allow building without widgets
-set(WebProcess_LIBRARIES
-    WebKit
+list(APPEND WebProcess_LIBRARIES
     Qt5::Widgets
     WebKitWidgets
 )

-set(NetworkProcess_LIBRARIES
-    WebKit
-)
-
-set(DatabaseProcess_LIBRARIES
-    WebKit
-)
-
-set(PluginProcess_LIBRARIES
-    WebKit
-)
-
 list(APPEND NetworkProcess_SOURCES
     NetworkProcess/EntryPoint/qt/NetworkProcessMain.cpp
 )
diff --git a/Source/WebKit2/UIProcess/API/qt/tests/CMakeLists.txt b/Source/WebKit2/UIProcess/API/qt/tests/CMakeLists.txt
index 899a9e038af..210342ef5bd 100644
--- a/Source/WebKit2/UIProcess/API/qt/tests/CMakeLists.txt
+++ b/Source/WebKit2/UIProcess/API/qt/tests/CMakeLists.txt
@@ -84,6 +84,12 @@ set(qmltests_LIBRARIES
     ${Qt5Test_LIBRARIES}
 )

+if (SHARED_CORE)
+    list(APPEND qmltests_LIBRARIES
+        WebKit2
+    )
+endif ()
+
 add_executable(tst_qmltests_WebView ${qmltests_SOURCES})
 target_compile_definitions(tst_qmltests_WebView PRIVATE ${tst_qmltests_WebView_DEFINITIONS})
 target_link_libraries(tst_qmltests_WebView ${qmltests_LIBRARIES})
diff --git a/Tools/MiniBrowser/qt/CMakeLists.txt b/Tools/MiniBrowser/qt/CMakeLists.txt
index e77d3bd5451..68dd13c8b70 100644
--- a/Tools/MiniBrowser/qt/CMakeLists.txt
+++ b/Tools/MiniBrowser/qt/CMakeLists.txt
@@ -53,6 +53,12 @@ set(MiniBrowser_LIBRARIES
     ${Qt5Quick_LIBRARIES}
 )

+if (SHARED_CORE)
+    list(APPEND MiniBrowser_LIBRARIES
+        WebKit2
+    )
+endif ()
+
 if (ENABLE_TEST_SUPPORT)
     add_definitions(-DHAVE_QTTESTSUPPORT)
 endif ()
```

**Comment by mkaracsony81** [comment]
I am trying to build it with lto enabled, which is why I need dynamic libraries instead of static ones. It may not even work, but it's worth a shot.

**Comment by annulen** [comment]
Ah, that certainly makes sense. Please use patch above

**Comment by annulen** [comment]
Did the patch help?

**Comment by mkaracsony81** [comment]
Oh sorry, yes it did. I was able to build it as shared, but as soon as I turned on LTO it failed. Undefined references again, but that's another issue.

**Comment by annulen** [comment]
Fixed with 261e80d9179327186ee59fd7ddbb4aa42a5d3fdb
